addons/editor-delete-button/userscript.js

```diff
@@ -0,0 +1,56 @@
+export default async ({ addon, msg }) => {
+  const { redux } = addon.tab;
+  // Wait until user has logged in, and is the author of the project
+  await redux.waitForState((state) => state.preview?.projectInfo?.author?.id === state.session?.session?.user?.id);
+  while (true) {
+    const loadItem = await addon.tab.waitForElement(
+      "div[class^='menu-bar_file-group'] > :nth-child(3) ul > :nth-child(4):not(.sa-editor-delete-button)",
+      {
+        markAsSeen: true,
+        reduxCondition: (state) => !state.scratchGui.mode.isPlayerOnly && !state.preview.visibilityInfo.deleted,
+      }
+    );
+
+    const dropdownItem = document.createElement("li");
+    dropdownItem.className = addon.tab.scratchClass("menu_menu-item", "menu_hoverable", {
+      others: ["sa-editor-delete-button"],
+    });
+    dropdownItem.innerText = msg("button-text");
+    addon.tab.displayNoneWhileDisabled(dropdownItem, { display: "block" });
```

Comment:
This has no effect if you don't add `"dynamicEnable": true` and `"dynamicDisable": true` to `addon.json`.